.github/actions/build-push/action.yml

```diff
@@ -79,3 +85,15 @@ runs:
         docker tag ${IMAGE_NAME} ghcr.io/${{ github.repository_owner }}/${IMAGE_NAME}
         docker push ghcr.io/${{ github.repository_owner }}/${IMAGE_NAME}
       shell: bash
+
+    - name: Write new image hash to file
+      if: ${{ inputs.write_image_hash == 'true' }}
+      run: |
+        git config user.name github-actions
+        git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+        git fetch && git checkout ci && git pull && git rebase origin/main
+        SHA=$(docker images --no-trunc --quiet ${IMAGE_NAME})
+        echo ${SHA} > ./${{ env.WORKSPACE_PATH }}/${{ env.OUTPUT_TAG }}-hash
+        git add ./${{ env.WORKSPACE_PATH }}/${{ env.OUTPUT_TAG }}-hash
+        git commit -m "Update ${{ env.IMAGE_NAME }} image hash" && git push -f
```

Comment:
In your fork, the CI branch has all SHA files in one commit with the message `Update ros2-modulo:galactic-devel image hash`. From seeing this build-push action I would have expected one commit per image (e.g. one commit per invocation of the build-push action from the workflow). Am I missing something?

Reply:
something is wrong indeed, thanks for pointing out...i have to investigate